sdcm/utils/data_validator.py

```diff
@@ -469,6 +499,171 @@ def list_of_view_names_for_update_test(self):
+            # data_for_validation.views[0] - view name with rows before update
+            # data_for_validation.views[1] - view name with rows after update
+            # data_for_validation.views[2] - view name with all expected partition keys
+            columns_for_validation = ', '.join(self.get_entity_columns(entity_name=data_for_validation.views[0],
+                                                                       session=session))
+            query = f"select %s f" \
+                    f"rom %s where {' and '.join(missed_row_as_str)}"
+
+            row_data = {}
+            for table in data_for_validation.views[:-1]:
+                if 'after_update' in table:
+                    key = 'after'
+                    select_columns = columns_for_validation
+                elif 'actual_data' in table:
+                    key = 'actual'
+                    select_columns = columns_for_validation
+                elif 'expect' in table:
+                    key = 'expect'
+                    select_columns = select_pk_columns
+                else:
+                    key = 'before'
+                    select_columns = columns_for_validation
+
+                try:
+                    row_data.update({key: list(session.execute(query % (select_columns, table)))})
+                except Exception as error:  # pylint: disable=broad-except
+                    LOGGER.error("Query %s failed. Error: %s", query % table, error)
+
+            row_data.update({'source_all_columns': list(session.execute(query % (columns_for_validation,
+                                                                                 self.base_table_name)))})
+            result[i].update({'rows': row_data})
+
+            if row_data['before'] or row_data['after']:
+                DataValidatorEvent.UpdatedRowsValidator(
+                    severity=Severity.WARNING,
+                    message=f"Row {diff_row} is found in the "
+                            f"{data_for_validation.views[0] if row_data['before'] else data_for_validation.views[1]} "
+                            f"view despite was not fetched. Test problem, failure in the fetcher\n"
+                            f"Row data in the source table: {row_data['source_all_columns']}\n"
+                            f"Row data in the expected table: {row_data['expect']}").publish()
+            else:
+                DataValidatorEvent.UpdatedRowsValidator(
+                    severity=Severity.ERROR,
+                    error=f"Row {diff_row} is expected to be find in whether before (the row was not updated) or "
+                    f"after (the row was updated) update view, but was not found.\n"
+                    f"Possible reason of the problem: "
+                    f"the row was updated but with not expected value as it defined in profile.\n"
+                    f"Row data in the source table: {row_data['source_all_columns']}").publish()
```

Comment:
same as the above, we can have the whole data inside an event.

Reply:
changed it to error event